### PR TITLE
add more command options and enhancment usage

### PR DESCRIPTION
**Pull request by Yancey0623**

### CODE REVIEW

File: src/transformers/trainer.py [hunk 1]
```diff
@@ -1972,6 +1981,9 @@ def training_step(self, model: nn.Module, inputs: Dict[str, Union[torch.Tensor,
         else:
             loss.backward()
 
+        if is_torch_ltc_available():
+            ltc.mark_step()
```
Comment (tanyokwok) [inline src/transformers/trainer.py:1984]: Why did the `mark_step()` location change to here? The previous locations are after `training_step` and `zero_grad`.

File: src/transformers/trainer.py [hunk 2]
```diff
@@ -1460,13 +1464,14 @@ def train(
                     f" num_steps ({max_steps}) higher than the number of available samples."
                 )
                 self.control.should_training_stop = True
-            if is_torch_disc_available():
+            if is_torch_ltc_available():
                 import torch._lazy.metrics as metrics
```
Comment (tanyokwok) [inline src/transformers/trainer.py]: The naming `metrics` here is the same as the `metrics` at: L1536

File: examples/torch_disc/torch_disc_train.sh [hunk 3]
```diff
@@ -4,26 +4,37 @@
...
-export PYTHONPATH=$(pwd)/../../src
-export CUDA_VISIBLE_DEVICES=1
-export LTC_DISC_CUDA=1
-#export BENCHMARK_ENABLE_NVPROF=1
-#export BENCHMARK_DISABLE_TORCH_DISC=1
-export LD_LIBRARY_PATH=/usr/local/cuda-11.0/targets/x86_64-linux/lib/:$LD_LIBRARY_PATH
-#nvprof --profile-from-start off python $train_script \
-python $train_script \
-  ../pytorch/text-classification/run_glue.py \
+function set_default_env() {
```
Comment (tanyokwok) [inline examples/torch_disc/torch_disc_train.sh:7]: Nit: shall we split training scripts and README.md into another pull request?

File: src/transformers/trainer.py (hunk 1 above)
Comment (Yancey0623) [reply src/transformers/trainer.py:1984]: call `ltc.mark_step` after the loss accumulation caused performance degradation(1.3x it/s to 1.1x it/s), I'm still looking for the reason, so I moved the `mark_step` in this pr.

File: examples/torch_disc/torch_disc_train.sh (hunk 3 above)
Comment (Yancey0623) [reply examples/torch_disc/torch_disc_train.sh:7]: I think it's okay, the README described the usage of the script, they are related.

File: src/transformers/trainer.py (hunk 2 above)
Comment (Yancey0623) [reply src/transformers/trainer.py]: done

**Review by tanyokwok — APPROVED** [review]
LGTM
